ui/src/frontend/hsluv_cache.ts

```diff
@@ -37,3 +37,25 @@ export function cachedHsluvToHex(
     hue: number, saturation: number, lightness: number): string {
   return cache.get(hue, saturation, lightness);
 }
+
+// A mapping of slice colors to contrasting colors
+// suitable for rendering text on them. Keys and
+// values are both hex codes of the form #rrggbb.
+const contrastingTextColorCodeCache = new Map<string, string>();
+
+// Obtain a color code contrasting to the given |color|
```

Comment:
Above and beyond! Thank you. I find these kinds of comments very helpful, but don't expect everyone to agree. I ask for a bare minimum when things are not self documenting. Thanks for going the extra mile